File: frontend/lib/categories.ts
export type CategoryDefinition = {
  name: string;
  colorClass: string;
  icon: string;
};

/**
 * Category definitions - fixed enum shared with backend.
 * Index = category code, value = { name, color class, icon }
 */
export const CATEGORIES: CategoryDefinition[] = [
  { name: "Sleep", colorClass: "cat-0", icon: "🌙" },
  { name: "Work", colorClass: "cat-1", icon: "💼" },
  { name: "Learning & Building", colorClass: "cat-2", icon: "📘" },
  { name: "Deep Thinking / Reflection", colorClass: "cat-3", icon: "🧠" },
  { name: "Exercise & Health", colorClass: "cat-4", icon: "🏋️" },
  { name: "Friends & Social", colorClass: "cat-5", icon: "🧑‍🤝‍🧑" },
  { name: "Relaxation & Leisure", colorClass: "cat-6", icon: "🎮" },
  { name: "Dating / Partner", colorClass: "cat-7", icon: "❤️" },
  { name: "Family", colorClass: "cat-8", icon: "👪" },
  { name: "Life Admin / Chores", colorClass: "cat-9", icon: "🧾" },
  { name: "Travel / Commute", colorClass: "cat-10", icon: "✈️" },
  { name: "Getting Ready / Misc", colorClass: "cat-11", icon: "🚿" },
];

export const CATEGORY_COUNT = CATEGORIES.length;

File: backend/app/api/endpoints/categories.py
"""Category reference endpoints (frontend is source of truth)."""

from fastapi import APIRouter

router = APIRouter(tags=["categories"])

# Category reference (for documentation, actual logic is frontend)
CATEGORIES: dict[int, str] = {
    0: "Sleep",
    1: "Work",
    2: "Learning & Building",
    3: "Deep Thinking / Reflection",
    4: "Exercise & Health",
    5: "Friends & Social",
    6: "Relaxation & Leisure",
    7: "Dating / Partner",
    8: "Family",
    9: "Life Admin / Chores",
    10: "Travel / Commute",
    11: "Getting Ready / Misc",
}


@router.get("/categories")
def get_categories():
    """
    Returns the category enum for reference.
    Frontend has this hardcoded, but useful for debugging/docs.
    """

    return CATEGORIES


